app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt

```diff
@@ -2251,6 +2251,118 @@ class PatientRepositoryAndroidTest {
         .inOrder()
   }
 
+  @Test
+  fun querying_all_patients_in_a_facility_should_set_the_the_latest_recorded_measurement_recorded_on_within_any_facility() {
+
+    fun createPatient(patientUuid: UUID, patientName: String): PatientProfile {
+      val patientProfile = testData.patientProfile(patientUuid = patientUuid).let { patientProfile ->
+        patientProfile.copy(patient = patientProfile.patient.copy(status = Active, fullName = patientName))
+      }
+      patientRepository.save(listOf(patientProfile)).blockingAwait()
+
+      return patientProfile
+    }
+
+    fun recordBp(uuid: UUID, patientUuid: UUID, facilityUuid: UUID, recordedAt: Instant) {
+      val bp = testData.bloodPressureMeasurement(uuid = uuid, patientUuid = patientUuid, facilityUuid = facilityUuid, recordedAt = recordedAt)
+
+      bloodPressureRepository.save(listOf(bp)).blockingAwait()
+    }
+
+    fun recordBloodSugar(uuid: UUID, patientUuid: UUID, facilityUuid: UUID, recordedAt: Instant) {
+      val bloodSugar = testData.bloodSugarMeasurement(uuid = uuid, patientUuid = patientUuid, facilityUuid = facilityUuid, recordedAt = recordedAt)
+
+      bloodSugarRepository.save(listOf(bloodSugar)).blockingAwait()
+    }
+
+    // given
+    val (facilityA, facilityB) = facilityRepository.facilities().blockingFirst()
+    val now = Instant.parse("2019-06-26T00:00:00Z")
+    val oneSecondLater = now.plusSeconds(1)
+    val oneSecondEarlier = now.minusSeconds(1)
+    val fiveSecondsLater = now.plusSeconds(5)
+
+    val uuidOfPatientA = UUID.fromString("e2a49529-3b46-4c9e-a7bc-af3090f1ecbb")
+    val uuidOfBp1OfPatientA = UUID.fromString("588b0d8c-6dce-4664-863f-8c79c06c6c3c")
+    val uuidOfBp2OfPatientA = UUID.fromString("ba875b98-4ca0-4097-bbc7-56041b377cf2")
+    val uuidOfBp3OfPatientA = UUID.fromString("4b92601e-4a86-40de-b6b7-707f9122a81b")
+    createPatient(patientUuid = uuidOfPatientA, patientName = "Patient with latest BP in Facility A")
+    recordBloodSugar(uuid = uuidOfBp1OfPatientA, patientUuid = uuidOfPatientA, facilityUuid = facilityA.uuid, recordedAt = oneSecondLater)
+    recordBp(uuid = uuidOfBp2OfPatientA, patientUuid = uuidOfPatientA, facilityUuid = facilityB.uuid, recordedAt = oneSecondEarlier)
+    recordBloodSugar(uuid = uuidOfBp3OfPatientA, patientUuid = uuidOfPatientA, facilityUuid = facilityA.uuid, recordedAt = now)
```

Comment:
Some of these are blood sugars and the others are BPs. Can we rename the variables to reflect that?